src/field/pokemon.ts

```diff
@@ -3581,26 +3554,18 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
     return false;
   }
 
-  /** @overload */
-  getTag(tagType: BattlerTagType): BattlerTag | nil;
-
-  /** @overload */
-  getTag<T extends BattlerTag>(tagType: AbstractConstructor<T>): T | nil;
-
-  getTag(tagType: BattlerTagType | AbstractConstructor<BattlerTag>): BattlerTag | nil {
+  getTag<T extends BattlerTag>(...tagTypes: BattlerTagType[]): T | nil {
     if (!this.summonData) {
       return null;
     }
-    return tagType instanceof Function
-      ? this.summonData.tags.find((t) => t instanceof tagType)
-      : this.summonData.tags.find((t) => t.tagType === tagType);
+    return this.summonData.tags.find((t) => tagTypes.includes(t.tagType)) as T | nil;
   }
 
-  findTag(tagFilter: (tag: BattlerTag) => boolean) {
+  findTag<T extends BattlerTag = BattlerTag>(tagFilter: (tag: BattlerTag) => boolean): T | nil {
     if (!this.summonData) {
       return null;
     }
-    return this.summonData.tags.find((t) => tagFilter(t));
+    return this.summonData.tags.find((t) => tagFilter(t)) as T | nil;
```

Comment:
```suggestion
    return this.summonData.tags.find((t) => tagFilter(t)) as T | undefined;
```
Same for this function.

Reply:
https://github.com/Despair-Games/poketernity/blob/67d278a1b60c8d878404c72948950559076811fe/src/field/pokemon.ts#L3589C1-L3593C74

Adopted from here